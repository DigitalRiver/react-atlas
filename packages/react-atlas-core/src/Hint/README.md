###### Default Hint:
    <div>
        <TextField small label="Phone Number:"/>
        <Hint>10 digit number including area code</Hint>
    </div>

###### Passing text as prop:
    <div>
        <TextField small label="CVV Code:"/>
	    <Hint text="The 3 or 4 digit number on the back of your card" />
    </div>


