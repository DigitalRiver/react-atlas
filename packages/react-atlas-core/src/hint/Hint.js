import React, { PropTypes } from "react";
import themeable from "react-themeable";

/**
 * Simple wrapper around a span to add 'hint'-like styles
 */
const Hint = ({ children, ...props }) => {
  const theme = themeable(props.theme);
  return <span {...props} {...theme(1, "base")}>{children}</span>;
};

Hint.defaultProps = {
  "theme": {
    "base": true
  }
};

Hint.propTypes = {
  "children": PropTypes.node,
  "theme": PropTypes.object
};

Hint.styleguide = {
  "category": "Form Components",
  "index": "3.5",
  "example": 
    `
<div>
  <p>Here is some regular text. <Hint>Some Hint text</Hint></p>
  <Input label="Here is a Label" placeholder="Placeholder Text" />
  <Hint>some Hint text below the input</Hint>
</div>
`
  
};

export default Hint;
