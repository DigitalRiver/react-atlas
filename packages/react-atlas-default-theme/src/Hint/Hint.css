.hintLabel {
	composes: dark-gray from '../styles.css';
	composes: italic from '../styles.css';
	composes: font-size-6 from '../styles.css';
}