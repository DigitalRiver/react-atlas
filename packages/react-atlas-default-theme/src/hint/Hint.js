import { Hint } from "react-atlas-core";
import style from "./hint.css";
import createComponent from "../createComponent";

export default createComponent(Hint, style);
