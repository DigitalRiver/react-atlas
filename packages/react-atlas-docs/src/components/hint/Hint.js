import React, { PropTypes } from "react";
import * as Atlas from 'react-atlas';

export default class Hint extends React.Component {
  constructor(props){
    super(props)
  }
  render() {
    return (
      <Atlas.Hint {...this.props}></Atlas.Hint>
    )
  }
}