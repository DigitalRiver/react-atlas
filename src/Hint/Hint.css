.hintLabel {
  	composes: default-font from '../styles.css';
	composes: dark-gray from '../styles.css';
	composes: font-size-6 from '../styles.css';
}