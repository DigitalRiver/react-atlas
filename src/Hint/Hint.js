import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Hint.css";

export class Hint extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const { className, children, text, style } = this.props;

    return text ? 
      <span style={style} className={cx(className)} styleName={"hintLabel"}>
        {text}
      </span>
     : 
      <span className={cx(className)} styleName={"hintLabel"}>
        {children}
      </span>
    ;
  }
}

Hint.propTypes = {
  /**
   * Text to be displayed can be passed as a child.
   * @examples '<Hint>This is a text hint</Hint>'
   */
  "children": PropTypes.node,
  /** An object, array, or string of CSS classes to apply to Hint.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   * Text that will be displayed (will override children if passed).
   * @examples '<Hint text="This is a text hint"/>'
   */
  "text": PropTypes.string,

  /**
   * Pass inline styling here.
   */
  "style": PropTypes.object
};

Hint.defaultProps = {
  "className": ""
};

export default CSSModules(Hint, styles, { "allowMultiple": true });
