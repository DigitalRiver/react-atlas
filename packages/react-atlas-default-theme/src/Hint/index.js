import * as HintStyle from "./Hint.css";
export { HintStyle };