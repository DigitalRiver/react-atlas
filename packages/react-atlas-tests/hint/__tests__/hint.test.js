import React from "react";
import { mount } from "enzyme";
import { default as Hint } from "../../../react-atlas-core/src/Hint/Hint";

describe("Test Hint component", () => {
  it("Test default props", function() {
    const result = mount(<Hint />);
  });
});
