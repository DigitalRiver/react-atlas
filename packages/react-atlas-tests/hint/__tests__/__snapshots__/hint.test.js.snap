// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hint component render Render correctly 1`] = `
<span
  className=""
  style={undefined}
  styleName="hintLabel"
>
  Some text
</span>
`;
